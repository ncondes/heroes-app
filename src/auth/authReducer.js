import { types } from '../types/types';

// const state = {
//     name: 'Nicolas',
//     logged: true
// }

// const loginAction = {
//     type: types.login,
//     payload: {
//         name: 'Nicolas',
//         email: 'Nicolas@gmail.com'
//     }
// }

export const authReducer = ( state = {}, action ) => {

    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true,
            }
        case types.logout:
            return {
                logged: false,
            }
    
        default:
            return state;
    }

}
