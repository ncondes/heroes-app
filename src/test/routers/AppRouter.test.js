import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/authContext';
import { mount } from 'enzyme';


describe('Pruebas en AppRouter', () => {

    const contextValue =  {
        user: {
            logged: false,
        },
    };

    test('Debe de mostrar el login screen si el usuario no esta autenticado', () => {

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h1').text().trim() ).toBe( 'Login' );

    })

    test('Debe de mostrar el marvel screen si el usuario esta autenticado', () => {
        
        const contextValue =  {
            user: {
                name: 'Rodrigo',
                logged: true,
            },
        };

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.navbar').exists() ).toBe( true );
        expect( wrapper.find('h1').text().trim() ).toBe( 'Marvel' );

    })
    
    
})