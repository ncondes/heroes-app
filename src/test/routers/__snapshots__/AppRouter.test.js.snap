// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe de mostrar el login screen si el usuario no esta autenticado 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <div
    className="d-flex justify-content-center"
  >
    <form
      className="col-12 col-sm-6"
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="username"
        >
          Username
        </label>
        <input
          autoComplete="off"
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Your username here"
          type="text"
        />
      </div>
      <div
        className="d-grid gap-2"
      >
        <button
          className="btn btn-outline-dark"
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Pruebas en AppRouter Debe de mostrar el marvel screen si el usuario esta autenticado 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="navbar-collapse collapse justify-content-between"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            className="nav-link false"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
          <a
            className="nav-link false"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            className="nav-link false"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
        <div
          className="navbar-nav"
        >
          <span
            className="nav-link text-info"
          >
            Rodrigo
          </span>
          <button
            className="nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Marvel
      </h1>
      <hr />
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-spider"
          onClick={[Function]}
        >
          <img
            alt="Spider Man"
            className="img img-responsive"
            src="./assets/marvel-spider.jpg"
          />
          <div
            className="profile-name"
          >
            Spider Man
          </div>
          <div
            className="profile-position"
          >
            Peter Parker
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Amazing Fantasy #15
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-captain"
          onClick={[Function]}
        >
          <img
            alt="Captain America"
            className="img img-responsive"
            src="./assets/marvel-captain.jpg"
          />
          <div
            className="profile-name"
          >
            Captain America
          </div>
          <div
            className="profile-position"
          >
            Steve Rogers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Captain America Comics #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-iron"
          onClick={[Function]}
        >
          <img
            alt="Iron Man"
            className="img img-responsive"
            src="./assets/marvel-iron.jpg"
          />
          <div
            className="profile-name"
          >
            Iron Man
          </div>
          <div
            className="profile-position"
          >
            Tony Stark
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Tales of Suspense #39
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-thor"
          onClick={[Function]}
        >
          <img
            alt="Thor"
            className="img img-responsive"
            src="./assets/marvel-thor.jpg"
          />
          <div
            className="profile-name"
          >
            Thor
          </div>
          <div
            className="profile-position"
          >
            Thor Odinson
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Journey into Myster #83
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-hulk"
          onClick={[Function]}
        >
          <img
            alt="Hulk"
            className="img img-responsive"
            src="./assets/marvel-hulk.jpg"
          />
          <div
            className="profile-name"
          >
            Hulk
          </div>
          <div
            className="profile-position"
          >
            Bruce Banner
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Incredible Hulk #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-wolverine"
          onClick={[Function]}
        >
          <img
            alt="Wolverine"
            className="img img-responsive"
            src="./assets/marvel-wolverine.jpg"
          />
          <div
            className="profile-name"
          >
            Wolverine
          </div>
          <div
            className="profile-position"
          >
            James Howlett
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Incredible Hulk #180
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-daredevil"
          onClick={[Function]}
        >
          <img
            alt="Daredevil"
            className="img img-responsive"
            src="./assets/marvel-daredevil.jpg"
          />
          <div
            className="profile-name"
          >
            Daredevil
          </div>
          <div
            className="profile-position"
          >
            Matthew Michael Murdock
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Daredevil #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-hawkeye"
          onClick={[Function]}
        >
          <img
            alt="Hawkeye"
            className="img img-responsive"
            src="./assets/marvel-hawkeye.jpg"
          />
          <div
            className="profile-name"
          >
            Hawkeye
          </div>
          <div
            className="profile-position"
          >
            Clinton Francis Barton
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Tales of Suspense #57
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-cyclops"
          onClick={[Function]}
        >
          <img
            alt="Cyclops"
            className="img img-responsive"
            src="./assets/marvel-cyclops.jpg"
          />
          <div
            className="profile-name"
          >
            Cyclops
          </div>
          <div
            className="profile-position"
          >
            Scott Summers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    X-Men #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-silver"
          onClick={[Function]}
        >
          <img
            alt="Silver Surfer"
            className="img img-responsive"
            src="./assets/marvel-silver.jpg"
          />
          <div
            className="profile-name"
          >
            Silver Surfer
          </div>
          <div
            className="profile-position"
          >
            Norrin Radd
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Fantastic Four #48
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;
